app/src/main/java/com/amaze/filemanager/activities/MainActivity.java

```diff
@@ -1321,7 +1321,7 @@ protected void onActivityResult(int requestCode, int responseCode, Intent intent
                         oppatheList.add(oppathe);
                         oppathe = "";
                     }
-                    for (int i = 0; i < oparrayListList.size(); i++) {
+                    for (int i = 0, size = oparrayListList.size(); i < size; i++) {
```

Comment:
Value in test expression get compared with every pass, so adding whole lot of time to calculate it again, to compare with current value of i. I think it's smart idea to initialize it instead of calculating it on the fly again and again unless required.

Reply:
Good and clear explanation

Reply:
Good and clear explanation

Reply:
> Value in test expression get compared with every pass, so adding whole lot of time to calculate it again, to compare with current value of i. I think it's smart idea to initialize it instead of calculating it on the fly again and again unless required.

What makes you think that the size has to be *calculated*? It is just returning a copy of an `int` (size field in `ArrayList`), that is really inexpensive.

Reply:
@EmmanuelMess Firstly you are a clover programmer! through calling `Arraylist.size()` won't  caculate size. But function call is expensive through this function just has a statement. I learn this skill from [okhttp](https://github.com/square/okhttp) source code

Reply:
> But function call is expensive through this function just has a statement.

What do you mean? Any further optimizations will probably be done by javabytecode pre-compiler.

Reply:
I think  optimize manully is a good choice, e.g Compiler may won't optimize on when run lower java version!